src/mixins/filing-template-mixin.ts

```diff
@@ -829,6 +833,10 @@ export default class FilingTemplateMixin extends DateMixin {
 
     // restore Restoration data
     this.setRestorationApprovalType(this.getStateFilingRestoration?.approvalType)
+    if (filing.restoration.approvalType) {
+      this.mutateRestorationApprovalType(filing.restoration.approvalType)
```

Comment:
Lines 835 and 837 look like duplicates (sort of).

Now the question is, which approval type needs to be restored here? Probably the second one if it exists, else the first one.

Reply:
Good call.  I've changed the logic as per your suggestion

Reply:
Thanks.

Is there any possiblity that line 838 is setting `undefined` into the store?

Reply:
I haven't changed the logic, `this.setRestorationApprovalType(this.getStateFilingRestoration?.approvalType)` is exactly as written previously.

Reply:
Yes, but I think it was a bug. Karim, thoughts?

Reply:
`this.setRestorationApprovalType(this.getStateFilingRestoration?.approvalType)` should work just fine! I've tested it and it'll never return `undefined`.

Reply:
Thanks @JazzarKarim and @severinbeauvais .  I'll leave this as is unless either of you object.

Reply:
OK to leave it as is. (That conditional chaining is bugging me though, lol.)